main.ml

```diff
@@ -1372,14 +1397,11 @@ type backend_error = string * (string list) [@@deriving sexp]
  *  printer we can log the actual arguments of the exception.
  * *)
 let register_exn_pretty_printers () =
-  Sexplib.Conv.Exn_converter.add ~finalise:false [%extension_constructor Storage_interface.Backend_error]
-  (function
-    | Storage_interface.Backend_error e -> sexp_of_backend_error e
-    | _ -> assert false);
-  Sexplib.Conv.Exn_converter.add ~finalise:false [%extension_constructor Storage_interface.Backend_error_with_backtrace]
-  (function
-    | Storage_interface.Backend_error_with_backtrace e -> sexp_of_backend_error e
-    | _ -> assert false)
+  Sexplib.Conv.Exn_converter.add ~finalise:false [%extension_constructor Storage_interface.Storage_error]
+    (function
+      | Storage_interface.Storage_error (Backend_error e) -> sexp_of_backend_error e
```

Comment:
Not sure if these are still needed here, you got pretty printers registered in xcp-idl, right?

Reply:
True, but these do seem to be more specific than my general registration. I'm inclined to leave them in place for now until we've got this PR merged.